frontend/shared/components/src/filters/filterBuilders.ts

```diff
@@ -196,93 +196,123 @@ export function getAdvancedMemberWithRegistrationsBlobUIFilterBuilders(platform:
-                                $gt: {$: '$now'}
-                            },
-                        }
-                    ])
-                }
-
-                if (choices.includes('Active') && !choices.includes('Expiring')) {
-                    filters.push(...[
-                        {
-                            expireDate: null,
-                            endDate: {
-                                $gt: {$: '$now'}
-                            },
-                        },
-                        {
-                            expireDate: {
-                                $gt: {$: '$now'}
-                            },
-                        }
-                    ])
-                }
-
-                if (!choices.includes('Active') && choices.includes('Expiring')) {
-                    filters.push(...[
-                        {
-                            expireDate: {
-                                $lt: {$: '$now'}
-                            },
-                            endDate: {
-                                $gt: {$: '$now'}
-                            },
-                        }
-                    ])
-                }
+                const choices = Array.isArray(f) ? f : [f];
 
-                if (choices.includes('Inactive')) {
-                    invertedFilters.push(...[
-                        {
-                            endDate: {
-                                $gt: {$: '$now'}
-                            },
-                        }
-                    ])
+                if(choices.length === 3) {
+                    return null;
                 }
 
-                const filter: StamhoofdFilter = []
-
-                if (filters.length > 0) {
-                    filter.push({
+                if(choices.length === 2 && ['Active', 'Expiring'].every(x => choices.includes(x))) {
+                    return {
                         platformMemberships: {
-                            $elemMatch: filters.length === 1 ? filters[0] : {
-                                $or: filters
+                            $elemMatch: {
+                                startDate: {
+                                    $lte: {$: '$now'}
+                                },
+                                $or: [
+                                    {
+                                        expireDate: null
+                                    },
+                                    {
+                                        expireDate: {
+                                            $gt: {$: '$now'}
+                                        }
+                                    }
+                                ],
+                                // just in case, maybe not necessary if expireDate is always less than endDate?
+                                endDate: {
+                                    $gt: {$: '$now'}
+                                }
```

Comment:
```suggestion
                                startDate: {
                                    $lte: {$: '$now'}
                                },
                                endDate: {
                                    $gt: {$: '$now'}
                                }
```

Als je zowel actieve als expired memberships toont, dan doet de expire date er niet toe, de filter klopte niet helemaal.